### PR TITLE
update hparams

### PR DESCRIPTION
**Pull request by fcuni**
Contribution to TODO 
> tune the hyperparameters so they are not terrible, I just winged it. (currently seeing val loss 2.06, recall count-based 4-gram was 2.11)

These choices improve valid loss to ~2.05. Swept over {learning_rate, weight_decay, batch_size, embedding_size and hidden_size} and found this combination to work best. See below for a plot with the current baseline vs the new setup,
image

[Link to wandb workspace 
](https://wandb.ai/fcunig/mlp_pytorch/workspace?nw=nwuserfcunig)
PS. if desired I can also provide the few lines of code to plot to WandB in a separate PR

### CODE REVIEW

**Comment by karpathy** [comment]
Very cool! :) 
Although... you basically made the network a lot larger :p is it a lot slower?
It's possible that at this stage we also introduce Dropout and then I think you can probably go even a few notches lower.

**Comment by karpathy** [comment]
Another thought... I wonder if I should merge it. It's kind of a fun exercise itself to optimize it :D

**Comment by fcuni** [comment]
> Although... you basically made the network a lot larger :p is it a lot slower?
yeh, i loaded the pytorch version to cuda but the numpy version was quite slow, see a plot below of the runtimes (y-axis is just some dummy variable in this case)

image

I was a annoyed that I could not make the loss lower at the original model size. so i went back over it today and double checked. attaching some plots of slicing of the loss landscape at fixed {embedding_size, hidden_size}, stepping (very coarsely) over {learning_rate, weight_decay, batch_size}. it seems like your "winging it" hparams are great choices at the original model size :) 

batch_size_64 0
batch_size_128 0
batch_size_256 0
batch_size_512 0

**Comment by fcuni** [comment]
> It's possible that at this stage we also introduce Dropout 
That would definitely help, i was playing under the constraint of not touching the model/optimiser to avoid messing up your project :) 

if you would like to introduce dropout, i am happy to help with that as well
